### DISCUSSION TITLE
Does Apple Podcasts really support the <podcast:transcript> tag?

### DISCUSSION THREAD

**Discussion by bertcoerver** [Podcastindex-org/podcast-namespace#617] [Clarification Needed]
Hello,

I've been experimenting a bit with the `transcript` tag (and podcasts in general) since iOS 17.4 came out, making a "private" podcast to be distributed through a simple URL pointing to a RSS feed. [Over-here](https://podcastindex.org/apps?elements=Transcript&platforms=iOS%2CmacOS) Apple Podcasts is listed as supporting the `<podcast:transcript>` tag and [these](https://help.apple.com/itc/podcasts_connect/#/itcb54353390) [two](https://podcasters.apple.com/support/5316-transcripts-on-apple-podcasts) documents from Apple say the same thing.

So I've got a [RSS feed](https://storage.googleapis.com/audiobooks-hmc/transcript_test/feed.rss), which has one episode for now. The episode has a `transcript` tag, which points to [this VTT file](https://storage.googleapis.com/audiobooks-hmc/transcript_test/transcripts/cbc032_04_various_128kb_32.vtt), and [this media file](https://storage.googleapis.com/audiobooks-hmc/transcript_test/cbc032_04_various_128kb.mp3) (an mp3).

When I run the feed through either the [PodBase Validator](https://podba.se/validate/?url=https://storage.googleapis.com/audiobooks-hmc/transcript_test/feed.rss) or the [Cast Feed Validator](https://www.castfeedvalidator.com/validate.php?url=https://storage.googleapis.com/audiobooks-hmc/transcript_test/feed.rss) it passes **all** tests and PodBase says it correctly detects the `transcript` tag. According to [Subtitle Validator](https://validator.subtitledpro.com), my VTT file is valid too.

However: 

- When I add the podcast to Apple Podcasts (through "Follow a Show by URL..."), the transcript doesn't show up (tested on MacOS and iOS).
- When I add the podcast to Apple Podcasts Connect ( Apple's platform to submit podcasts to its own "podcast index"), the transcript doesn't show up either (I'm talking to their support about this too), also see the screenshot below.

So if neither Apple Podcasts nor Apple Podcasts Connect detects or does something with the `<podcast:transcript>` tag, does it really "support" it? Does anyone have any ideas or suggestions on how to make this work (I'm very new to all this, so maybe I'm just doing something wrong)? Does anyone have any examples of a podcast feed that uses the tag successfully (so not just the auto-generated transcripts)?

Regards,
Bert

<img width="810" alt="Screenshot 2024-03-13 at 09 42 23" src="https://github.com/Podcastindex-org/podcast-namespace/assets/18460767/304932ed-0d96-4365-9076-9b0cf1e0b5e2">

**Comment by jamescridland** [comment]
Hi, Bert. I feel you! I couldn't get it to work for a bit too. The VTT file you've posted is also the correct content-type leaving the server, which is good.

I found that Apple Podcasts can take 24 hours to realise that you have a transcript, and to parse it. Then it'll appear in that window.

A few feeds that *are* definitely using the supplied `podcast:transcript` tag are:

* The [Podnews Weekly Review](https://podnews.net/podcast/i5hv2), which is hosted by Buzzsprout. That's an SRT file - they don't produce VTT files.
* The [Podnews Daily](https://podnews.net/podcast/i8xe9), which is closer to yours - it's hosted by me on Amazon S3. I've been using a SRT file over the past week, hoping it would work better than the VTT file that I have been feeding it: it seems to make little difference.

For the Podnews Daily, here it is - you're right, it appears in that window.

<img width="672" alt="Screenshot 2024-03-13 at 5 27 12 pm" src="https://github.com/Podcastindex-org/podcast-namespace/assets/231941/5a4f2f8e-150b-46cd-99ae-dbd84a90fb88">

Unfortunately, I can't get any feedback from Apple as to why they sometimes don't work, or why they do. It would be helpful if they were to feed back any errors. It *could* be that the timing is out, and therefore it's trying to match the words and failing. Or it could be something else.

I don't feel I'm helping much, but at least I can show you some that *do* work.

(Heads-up: I'm going to revert back to using VTT shortly for the Podnews Daily, so this example won't always be entirely correct).

**Reply by bertcoerver** [reply]
Thanks for your reply @jamescridland! Definitely useful to know that others are struggling with this too and to have some examples of it actually working, although my final goal is to have the transcript show up in Apple Podcasts without using Apple Podcasts Connect.

I'll continue doing some tests, like downloading the "auto-generated" VTT file and feeding that back into the `transcript` tag, assuming that Apple's own transcripts should pass their own tests, right? 🥴

**Reply by jamescridland** [reply]
> my final goal is to have the transcript show up in Apple Podcasts without using Apple Podcasts Connect

I don't use Apple Podcasts Connect for anything (and in any case, it only lets you VIEW, not EDIT anything - the beguiling EDIT button there is just a settings button).

> downloading the "auto-generated" VTT file and feeding that back into the transcript tag, assuming that Apple's own transcripts should pass their own tests, right?

Now that the VTT file Apple produces is actually compliant, yes. ;)

But - I am vaguely wondering if Apple needs to allowlist your podcast for transcripts, if you're hosting yourself. I do know that I couldn't get Apple to "see" my local transcripts for some time. I'd encourage you to contact Apple Support with your questions, because the more people who do that, the better it will be.

Finally - I'd also recommend using a compliant service like Podverse, Podfriend, PodcastAddict, etc, to view transcripts in the app.

**Reply by bertcoerver** [reply]
@jamescridland Thanks for all the advice! About the compliant services, would that allow me to have a private podcast (i.e. one that no-one else can find in some index)? I would like to listen to audiobooks that I own and read along with a transcript (that would basically be my dream-tool for language learning 😊).

**Reply by jamescridland** [reply]
> I would like to listen to audiobooks that I own and read along with a transcript

My thoughts:
1. Upload it to YouTube as a private file... then turn on subtitles when you play it back...
2. Use VLC to [burn a video of an audio file with transcripts](https://gotranscript.com/blog/how-to-hardcode-subtitles-with-vlc-media-player)
3. Or just use the AUDIO tag on a (password-protected) webpage. Look at the example at the bottom of https://podnews.net/article/apple-podcasts-transcriptions-faq :)

**Reply by bertcoerver** [reply]
Hehe yeah I know there are many ways to do this, right now I just listen to the audio and read along in the book, which works but is not ideal. I just really like the way Apple Podcasts shows the transcripts (previewing upcoming text, not moving constantly but only scrolling when a paragraph has finished and it just looks really slick) and also like to learn something new trying this 🤓.

**Comment by theDanielJLewis** [comment]
Apple Podcasts _should_ support the transcript when you add a feed directly, but I might not be surprised if they actually don't.

However, on the Podcasts Connect side, you have to go into the podcast's settings and change the default from using Apple's transcripts to using your provided ones.
